earth_enterprise/src/fusion/autoingest/plugins.src/RasterProject.src

```diff
@@ -1132,20 +1123,23 @@ uint32 RasterProjectAssetVersionImplD::BuildIndex(
     PacketGenExtraUpdateArg<RasterProductAssetVersion> extraArg;
     extraArg.insetInfo.reserve(neededIndexes.size());
     extraArg.genInfo.reserve(neededIndexes.size());
-    for (std::vector<uint>::const_iterator idx = neededIndexes.begin();
-         idx != neededIndexes.end(); ++idx) {
-      extraArg.insetInfo.push_back(&inset_infos[*idx]);
-      extraArg.genInfo.push_back(&genInfo[*idx]);
+    extraArg.calc = calc;
+
+    for(const auto& idx : neededIndexes) {
+      extraArg.insetInfo.push_back(&inset_infos[idx]);
+      extraArg.genInfo.push_back(&genInfo[idx]);
     }
 
-    // reuse or make a new PacketGenAssetVersion
-    MutableAssetVersionD packgen =
-      PacketGenFactory::ReuseOrMakeAndUpdate(genInfo[i].assetname,
-                                             type,
-                                             khMetaData(),
-                                             packetGenConfig,
-                                             extraArg);
+    // reuse or make a new PacketGenAssetVersion --> update children
+    // refactor this method call if wanting to do all calculations at once
```

Comment:
Minor: I think we can remove this comment.

Also minor: I think the extra indentation was added below by accident.